classes/table/timeslots/timeslots_table.php

```diff
@@ -106,24 +108,41 @@ public function col_action($row) {
         switch($this->displaymode) {
             case \mod_observation\table\timeslots\timeslots_display::DISPLAY_MODE_EDITING:
                 $htmlout .= \mod_observation\table\common::action_button(new \moodle_url($this->baseurl,
-                    ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'edit']), get_string('edit', 'observation'));
+                    ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'edit', 'sesskey' => sesskey()]),
+                    get_string('edit', 'observation'), 'btn-secondary');
                 $htmlout .= \mod_observation\table\common::action_button(new \moodle_url($this->baseurl,
-                    ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'delete']), get_string('delete', 'observation'));
-            break;
+                    ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'delete', 'sesskey' => sesskey()]),
+                    get_string('delete', 'observation'), 'btn-secondary');
+
+                if ($row->observee_id !== null) {
+                    $htmlout .= \mod_observation\table\common::action_button(new \moodle_url($this->baseurl,
+                        ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'kick', 'sesskey' => sesskey()]),
+                        get_string('kickobservee', 'observation'), 'btn-warning');
+                }
+                break;
 
             case \mod_observation\table\timeslots\timeslots_display::DISPLAY_MODE_SIGNUP:
                 $htmlout .= \mod_observation\table\common::action_button(new \moodle_url($this->baseurl,
-                    ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'join']), get_string('join', 'observation'));
+                    ['id' => $row->obs_id, 'slotid' => $row->id, 'action' => 'join', 'sesskey' => sesskey()]),
+                    get_string('join', 'observation'), 'btn-secondary');
             break;
 
             case \mod_observation\table\timeslots\timeslots_display::DISPLAY_MODE_UPCOMING:
                 if ($row->observee_id !== null) {
                     $htmlout .= \mod_observation\table\common::action_button(new \moodle_url($this->baseurl, ['id' => $row->obs_id,
-                        'slotid' => $row->id, 'action' => 'startsession']), get_string('startobservationsession', 'observation'));
+                        'slotid' => $row->id, 'action' => 'startsession', 'sesskey' => sesskey()]),
+                        get_string('startobservationsession', 'observation'), 'btn-secondary');
                 } else {
                     $htmlout .= get_string('noobservee', 'observation');
                 }
             break;
+            case \mod_observation\table\timeslots\timeslots_display::DISPLAY_MODE_OBSERVEE_REGISTERED:
+                if (\mod_observation\timeslot_manager::can_unenrol($row->obs_id, $row->id, $row->observee_id, false) === true) {
```

Comment:
`if ($condition === true)`  can basically always be just `if($condition)` unless you are trying to ensure there isnt a type coercion from something like string -> int, but in that case I feel like the code would be bugprone either way :). In your case, just a little anti-pattern to watch out for.

In the same vein `if ($var) {return true;} else {return false;}` is one that is common, and can reduce all the way down to `return $var` which people often overlook